core/src/main/java/org/axonframework/queryhandling/responsetypes/MultipleInstancesResponseType.java

```diff
@@ -56,10 +56,11 @@ public MultipleInstancesResponseType(Class<R> expectedCollectionGenericType) {
      */
     @Override
     public boolean matches(Type responseType) {
-        return isIterableOfExpectedType(responseType) ||
-                isStreamOfExpectedType(responseType) ||
-                isGenericArrayOfExpectedType(responseType) ||
-                isArrayOfExpectedType(responseType);
+        Type unwrapped = unwrapIfTypeFuture(responseType);
```

Comment:
Might be nice to add a test case to the `MultipleInstancesResponseTypeTest` for when a `@QueryHandler` annotated function returns a `Future`/`CompletableFuture`. Mainly stating this as the `FutureAsResponseTypeToQueryHandlersTest.java` you've added tests the whole working together, whilst I think testing this single unit would be nice as well.